bin/cli.mjs

```diff
@@ -1,161 +1,62 @@
- * @property {boolean} useGit Specifies whether the parser should execute optional git commands. (Should only be used within a git repo)
- * @property {keyof reporters} reporter Specifies the linter reporter.
- *
- * @name ProgramOptions
- * @type {Options}
- * @description The return type for values sent to the program from the CLI.
- */
-const {
-  input,
-  ignore,
-  output,
-  target = [],
-  version,
-  changelog,
-  disableRule,
-  lintDryRun,
-  gitRef,
-  reporter,
-  threads,
-} = program.opts();
+// Registering generate and lint commands
+commands.forEach(({ name, description, options, action }) => {
+  const cmd = program.command(name).description(description);
 
-const linter = createLinter(lintDryRun, disableRule);
+  // Add options to the command
+  Object.values(options).forEach(({ flags, desc, prompt }) => {
+    const option = new Option(flags.join(', '), desc).default(
+      prompt.initialValue
+    );
 
-const { loadFiles } = createMarkdownLoader();
-const { parseApiDocs } = createMarkdownParser();
+    if (prompt.required) {
+      option.makeOptionMandatory();
+    }
 
-const apiDocFiles = await loadFiles(input, ignore);
+    if (prompt.type === 'multiselect') {
+      option.choices(prompt.options.map(({ value }) => value));
+    }
 
-const parsedApiDocs = await parseApiDocs(apiDocFiles);
+    cmd.addOption(option);
+  });
 
-const { runGenerators } = createGenerator(parsedApiDocs);
+  // Set the action for the command
+  cmd.action(action);
+});
 
-// Retrieves Node.js release metadata from a given Node.js version and CHANGELOG.md file
-const { getAllMajors } = createNodeReleases(changelog);
+// Register the interactive command
+program
+  .command('interactive')
+  .description('Launch guided CLI wizard')
+  .action(interactive);
 
-// Runs the Linter on the parsed API docs
-linter.lintAll(parsedApiDocs);
+// Register the list command
+program
+  .command('list')
+  .addArgument(new Argument('<types>', 'The type to list').choices(types))
+  .description('List the given type')
+  .action(list);
 
-if (target) {
-  await runGenerators({
-    // A list of target modes for the API docs parser
-    generators: target,
-    // Resolved `input` to be used
-    input: input,
-    // Resolved `output` path to be used
-    output: output && resolve(output),
-    // Resolved SemVer of current Node.js version
-    version: coerce(version),
-    // A list of all Node.js major versions with LTS status
-    releases: await getAllMajors(),
-    // An URL containing a git ref URL pointing to the commit or ref that was used
-    // to generate the API docs. This is used to link to the source code of the
-    gitRef,
-    // How many threads should be used
-    threads,
+// Register the help command
+program
+  .command('help [cmd]')
+  .description('Show help for a command')
+  .action(cmdName => {
+    const target = program.commands.find(c => c.name() === cmdName) ?? program;
```

Comment:
:thinking:

Reply:
Commander.js does it already, I didn't know that 😃